homeassistant/components/synology_dsm/binary_sensor.py

```diff
@@ -82,7 +87,7 @@ class SynoDSMStorageBinarySensor(SynologyDSMDeviceEntity, BinarySensorEntity):
     @property
     def is_on(self) -> bool:
         """Return the state."""
-        return getattr(self._api.storage, self.entity_type)(self._device_id)
+        return bool(getattr(self._api.storage, self.entity_type)(self._device_id))
```

Comment:
Here `getattr(self._api.storage, self.entity_type)` returns `Callable` but in `SynoDSMUpgradeBinarySensor` `getattr(self._api.upgrade, self.entity_type)` returns bool? That's a weird api.